src/index.ts

```diff
@@ -45,37 +63,71 @@ const isExecutable = async (abspath: string, opt: LookPathOption = {}): Promise<
  * @param {LookPathOption} opt Options for lookpath.
- * @return {Promise<string|undefined>} Resolves absolute file path, or undefined if not found.
+ * @return {Promise<string|undefined | (string|undefined)[]>} Resolves absolute file path, or undefined if not found.
  */
-export async function lookpath(command: string, opt: LookPathOption = {}): Promise<string | undefined> {
-
+// Stronger typed functions so that it can infer if it's returning an array with opt.findAll or not
+export function lookpath(command: string): Promise<MaybeString>;
+export function lookpath(command: string, opt: LookOptionsFindAll): Promise<MaybeString[]>;
+export function lookpath(command: string, opt: BaseLookPathOption | LookOptionsNoFindAll): Promise<MaybeString>;
+export async function lookpath(command: string, opt: LookPathOption = {}): Promise<MaybeString | MaybeString[]> {
     const directpath = isFilepath(command);
     if (directpath) return isExecutable(directpath, opt);
 
     const dirs = getDirsToWalkThrough(opt);
-    const bins = await Promise.all(dirs.map(dir => isExecutable(path.join(dir, command), opt)));
-    return bins.find(bin => !!bin);
+    const bins = await Promise.all(dirs.map((dir) => isExecutable(path.join(dir, command), opt)));
+    return opt.findAll ? Array.from(new Set(bins.filter(Boolean).flat())) : bins.find((bin) => !!bin);
 }
 
+/**
+ * Synchronous version with absolute file path of given command,
+ * and returns undefined if the command not found.
+ * @param {string} command Command name to look for.
+ * @param {LookPathOption} opt Options for lookpath.
+ * @return {string|undefined | (string|undefined)[]} Resolves absolute file path, or undefined if not found.
+ */
+export function lookpathSync(command: string): MaybeString;
+export function lookpathSync(command: string, opt: LookOptionsFindAll): MaybeString[];
+export function lookpathSync(command: string, opt: BaseLookPathOption | LookOptionsNoFindAll): MaybeString;
+export function lookpathSync(command: string, opt: LookPathOption = {}): MaybeString | MaybeString[] {
+    const directpath = isFilepath(command);
+    if (directpath) return isExecutableSync(directpath, opt);
+
+    const dirs = getDirsToWalkThrough(opt);
+    const bins = dirs.map((dir) => isExecutableSync(path.join(dir, command), opt));
+    return opt.findAll ? Array.from(new Set(bins.filter(Boolean).flat())) : bins.find((bin) => !!bin);
+}
 /**
  * Options for lookpath.
  */
-export interface LookPathOption {
+interface BaseLookPathOption {
     /**
      * Additional pathes to look for, would be dealt same as PATH env.
      * Example: ['/tmp/bin', 'usr/local/bin']
      */
     include?: string[];
+    /**
+     * Look in common paths if they're not included in process.env or opts.env
+     * Those common paths are enumerated in `src/common.ts`
+     */
+    includeCommonPaths?: boolean;
+    /**
+     * Returns an array of all binaries matching that name in the PATH.
+     */
+    findAll?: boolean;
```

Comment:
You are right. Users might want to know all the matching bins sometimes.
But I prefer to export another function like `lookpathAll` or something to provide that capability, rather than adding an option.